src/App.tsx

```diff
@@ -8,37 +8,130 @@ import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
+import { getUsers } from './api/users';
+import { User } from './types/User';
+import { getPosts } from './api/posts';
+import { Post } from './types/Post';
 
 export const App: React.FC = () => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+  const [isErrorPosts, setIsErrorPosts] = useState(false);
+  const [posts, setPosts] = useState<Post[] | null>(null);
```

Comment:
you could use an empty array instead of null

Reply:
Good point